### ISSUE TITLE
Deprecate Lore

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6907]
|[<img alt="khorn's avatar" src="https://avatars.githubusercontent.com/u/815352?s=50" width="50" height="50">](https://github.com/khorn)| @khorn reported|
|-|-|
|Trac ID|trac#6907|
|Type|enhancement|
|Created|2014-01-17 16:19:05Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-lore-6907|

As Twisted has switched to using Sphinx for generating documentation, Lore is no longer needed, and should be deprecated.

Once Lore has been deprecated, and is no longer needed, another ticket should be created to remove it.

See: http://twistedmatrix.com/trac/wiki/CompatibilityPolicy#Modules

<details><summary>Searchable metadata</summary>

```
trac-id__6907 6907
type__enhancement enhancement
reporter__khorn khorn
priority__normal normal
milestone__Twisted_14_0_0 Twisted-14.0.0
branch__branches_deprecate_lore_6907 branches/deprecate-lore-6907
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__lore lore
keywords__ 
time__1389975545000000 1389975545000000
changetime__1393683829000000 1393683829000000
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [41772]) Branching to deprecate-lore-6907.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

I've taken a stab at deprecating it.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban commented|
|-|-|

Looks good to me. Thanks! Buildbot test are not available.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun set owner to @hawkowl|
|-|-|

Thanks.

  1. I'm not sure I see the benefit of deleting the _TODO_ comment in `twisted/lore/__init__.py` as part of this change.
  1. I've tried to figure out why this module uses `unittest.TestCase` but I can't.  I don't think the module should use *both* `unittest.TestCase` *and*` twisted.trial.unittest.TestCase`.  One thing I do see is that the existing twisted imports in this test module are a subset of the imports in `twisted/__init__.py`.  I don't see how that could actually matter in this case though.  I suspect since all the tests pass even with the new `twisted.trial.unittest` import it's safe to switch this code entirely over to trial.  Also, please use the `from twisted.trial.unittest import TestCase` style of import.
  1. You might like `namedAny("twisted.lore")` or maybe just `__import__("twisted.lore")` to avoid needing to hack around pyflakes warnings.
  1. American English spelling, sorry!  _favor_ instead of _favour_.
  1. There's no standard (maybe there should be) for this but consider:
```
exarkun@top:~/Projects/Twisted/trunk$ grep "was deprecated" twisted/ -r --include '*.py' | wc -l
76
exarkun@top:~/Projects/Twisted/trunk$ grep "is deprecated" twisted/ -r --include '*.py' | wc -l
120
```

Thanks again.  Please merge after addressing those points.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl set status to `closed`|
|-|-|

(In [41790]) Merge deprecate-lore-6907: Deprecate Twisted Lore

Author: hawkowl
Reviewer: exarkun
Fixes: [#6907](https://github.com/twisted/twisted/issues/6907)

This deprecates Twisted Lore, in favor of using Sphinx, from Twisted 14.0.
